cmd/kubelet/app/server.go

```diff
@@ -1243,6 +1251,60 @@ func parseResourceList(m map[string]string) (v1.ResourceList, error) {
 	return rl, nil
 }
 
+func parsePreReservedMemoryConfig(config []map[string]string) (map[int]map[v1.ResourceName]resource.Quantity, error) {
+	if len(config) == 0 {
+		return nil, nil
+	}
+
+	const (
+		indexKey = "numa-node"
+		typeKey = "memory-type"
+		limitKey = "limit"
+	)
+
+	keys := []string{indexKey, typeKey, limitKey}
+
+	// check whether all keys are present
+	for _, m := range config {
+		for _, key := range keys {
+			if _, exist := m[key]; !exist {
+				return nil, fmt.Errorf("key: %s is missing in given PreReservedMemoryZone flag: %v", key, config)
+			}
+		}
+	}
+
+	parsed := make(map[int]map[v1.ResourceName]resource.Quantity, len(config))
+	for _, m := range config {
+		idxInString, _ := m[indexKey]
+		idx, err := strconv.Atoi(idxInString)
+		if err != nil || idx < 0  {
+			return nil, fmt.Errorf("NUMA index conversion error for value: \"%s\"", idxInString)
+		}
+
+		typeInString, _ := m[typeKey]
+		v1Type := v1.ResourceName(typeInString)
+		if v1Type != v1.ResourceMemory && !corev1helper.IsHugePageResourceName(v1Type) {
+			return nil, fmt.Errorf("memory type conversion error, unknown type: \"%s\"", typeInString)
+		}
+		if corev1helper.IsHugePageResourceName(v1Type) {
+			if _, err := corev1helper.HugePageSizeFromResourceName(v1Type); err != nil {
+				return nil, fmt.Errorf("memory type conversion error, unknown type: \"%s\"", typeInString)
+			}
+		}
+
+		limitInString, _ := m[limitKey]
+		limit, err := resource.ParseQuantity(limitInString)
+		if err != nil || limit.Sign() != -1 {
+			return nil, fmt.Errorf("memory limit conversion error for value \"%s\"", limitInString)
```

Comment:
Without minus? `limit.Sign() != 1`

Reply:
you are right, I think I got destracted by something.
```
// Sign returns 0 if the quantity is zero, -1 if the quantity is less than zero, or 1 if the
// quantity is greater than zero.
func (q *Quantity) Sign() int {
```

Reply:
Done